pkg/tracing/tracing.go

```diff
@@ -118,13 +120,15 @@ func SetupTracing(ctx context.Context, name string, c Config) (func(context.Cont
 func newTraceProvider(name string, c Config, exporter *otlptrace.Exporter) *sdktrace.TracerProvider {
 	options := []sdktrace.TracerProviderOption{
 		sdktrace.WithBatcher(exporter),
-		sdktrace.WithSampler(sdktrace.ParentBased(sdktrace.TraceIDRatioBased(c.Otel.SampleRatio))),
 		sdktrace.WithResource(newResource(name)),
 	}
 
 	switch strings.ToLower(c.Otel.ExporterType) {
 	case "awsxray":
 		options = append(options, sdktrace.WithIDGenerator(xray.NewIDGenerator()))
+		options = append(options, sdktrace.WithSampler(sdktrace.ParentBased(sampler.NewRandTraceIDRatioBased(c.Otel.SampleRatio, rand.New(rand.NewSource(rand.Int63()))))))
```

Comment:
```suggestion
		options = append(options, sdktrace.WithSampler(sdktrace.ParentBased(sampler.NewRandTraceIDRatioBased(c.Otel.SampleRatio, rand.New(rand.NewSource(time.Now()))))))
```

Not sure it's advisable to seed a random source with a "random" value.